partner_credit_limit/migrations/11.0.0.0.2/post-migration.py

```diff
@@ -0,0 +1,28 @@
+from odoo import api, SUPERUSER_ID
+
+
+def migrate(cr, version):
+    set_credit_limit_values_to_partner(cr)
+
+
+def set_credit_limit_values_to_partner(cr):
+    env = api.Environment(cr, SUPERUSER_ID, {})
+    env.cr.execute(
+        """
+        SELECT id, credit_limit, grace_payment_days FROM res_partner
+        WHERE credit_limit != 0 OR grace_payment_days != 0""")
+    data = env.cr.dictfetchall()
+    credit_limit_vals = {
+        p['id']: {
+            'credit_limit': p['credit_limit'],
+            'grace_payment_days': p['grace_payment_days']} for p in data}
+    partner_ids = [p.get('id') for p in data]
```

Comment:
```python
    env = api.Environment(cr, SUPERUSER_ID, {})
    env.cr.execute(
        """
        SELECT id, credit_limit, grace_payment_days FROM res_partner
        WHERE credit_limit != 0 OR grace_payment_days != 0""")
    data = env.cr.fetchall()
    for company in env['res.company'].search([]):
        for partner_id, credit_limit, grace_payment_days in data:
            env['res.partner].browse(partner_id).with_context(force_company=company.id).write({
                'credit_limit': credit_limit
                'grace_payment_days': grace_payment_days})
```

Reply:
I like the browse out the for loop, should be faster ?

this is just a migration script, that will be executed only one time

Reply:
The browse is faster to fetch data.
But for write data it is the same.
I know this will be executed just one time
But all PM must understand the full change
And the current code is so heavy to understand using too much transformations